core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/aggregation/AvgAggregator.java

```diff
@@ -63,19 +62,19 @@ public String toString() {
   /**
    * Average State.
    */
-  protected class AvgState implements AggregationState {
+  protected static class AvgState implements AggregationState {
     private int count;
     private double total;
-    private boolean isNullResult = false;
+    private boolean isEmptyCollection = true;
 
-    public AvgState() {
+    AvgState() {
       this.count = 0;
       this.total = 0d;
     }
 
     @Override
     public ExprValue result() {
-      return isNullResult ? ExprNullValue.of() : ExprValueUtils.doubleValue(total / count);
+      return isEmptyCollection ? ExprNullValue.of() : ExprValueUtils.doubleValue(total / count);
```

Comment:
Can this `isEmptyCollection` be replaced by `count == 0`? Because I notice `count` already represents the count of non-null/missing value only. Same question for all other aggregators.

Reply:
made this change in avg/count/sum, but for min and max, we don't have to count the number so still using `isEmptyCollection`.